typescript/testsuite/network_impl/test_network.ts

```diff
@@ -0,0 +1,350 @@
+import { APIClient } from "../api/api_service_client/api_client";
+import { DatastoreClient } from "../datastore/datastore_service_client/datastore_client";
+import { ServiceID, NetworkContext, ContainerCreationConfig, StaticFileID, ContainerRunConfig, ContainerCreationConfigBuilder, ContainerRunConfigBuilder, ServiceContext, PortBinding } from "kurtosis-core-api-lib";
+import { Result, ok, err, ResultAsync, okAsync, errAsync } from "neverthrow";
```

Comment:
So I dug into what the `ResultAsync` class really is - it's a convenience around `Promise<Result<...>>`. IMO we shouldn't use it at all, since `Promise<Result<..>>` is way clearer and an `async` function requires you to return a `Promise` anyways (`ResultASync` won't work). So let's abolish `okAsync` and `errAsync` from our codebase, in favor of just regular `Result`